wallet-test/src/test/scala/org/bitcoins/wallet/LegacyWalletTest.scala

```diff
@@ -0,0 +1,39 @@
+package org.bitcoins.wallet
+
+import org.bitcoins.wallet.api.UnlockedWalletApi
+import org.bitcoins.wallet.util.BitcoinSWalletTest
+import org.scalatest.FutureOutcome
+import org.bitcoins.wallet.api.UnlockWalletError.BadPassword
+import org.bitcoins.wallet.api.UnlockWalletError.JsonParsingError
+import org.bitcoins.wallet.api.UnlockWalletSuccess
+import org.bitcoins.core.crypto.AesPassword
+import org.bitcoins.wallet.api.UnlockWalletError.MnemonicNotFound
+import com.typesafe.config.ConfigFactory
+import org.bitcoins.core.protocol.P2PKHAddress
+
+class LegacyWalletTest extends BitcoinSWalletTest {
+
+  override type FixtureParam = UnlockedWalletApi
+
+  override def withFixture(test: OneArgAsyncTest): FutureOutcome = {
+    val confOverride =
+      ConfigFactory.parseString("bitcoin-s.wallet.defaultAccountType = legacy")
+    withNewConfiguredWallet(confOverride)(test)
+  }
+
+  // eventually this test should NOT succeed, as BIP44
+  // requires a limit to addresses being generated when
+  // they haven't received any funds
```

Comment:
20 is the limit right? Perhaps add that. 

But I thought our wallets for the purposes of unit tests we have in memory databases? So every new run we create a new table which means we don't actually store our state to disk? Which means we shouldn't run into this limit?

Reply:
The `eventually` here refers to "once we get around to implementing that part of the BIP". But you're also right about using in-memory databases. I think I'll remove that comment, as it is slightly misleading.